presto-feature-toggle/README.md

```diff
@@ -0,0 +1,233 @@
+                .enabled(true)
+                .toggleStrategy("AllowList")
+                .registerToggleStrategy("AllowList", AllowListToggleStrategy.class)
+                .toggleStrategyConfig(ImmutableMap.of("allow-list-source", ".IDEA.", "allow-list-user", ".*prestodb"))
+                .bind();
+        }
+    }
+```
+
+Feature toggle definition with already registered toggle strategy.
+
+```
+    public class RegisterStrategyModule
+        implements Module
+    {
+        @Override
+        public void configure(Binder binder)
+        {
+            featureToggleBinder(binder)
+                .featureId("query-cancel")
+                .enabled(true)
+                .toggleStrategy("AllowList")
+                .toggleStrategyConfig(ImmutableMap.of("allow-list-source", ".IDEA.", "allow-list-user", ".*prestodb"))
+                .bind();
+        }
+    }
+```
+
+Toggle strategy configuration params can be updated on runtime, by changing strategy configuration param values.
+
+In this case, we can change `feature.query-cancel.strategy.allow-list-source` and `feature.query-cancel.strategy.allow-list-user` param values.
+
+# Examples
+
+## Query Cancel Feature
+
+```
+    feature.query-cancel.enable=true
+    feature.query-cancel.strategy=AllowList
+    feature.query-cancel.strategy.allow-list-source=.*IDEA.*
+    feature.query-cancel.strategy.allow-list-user=.*prestodb
+```
+
+Feature Toggle Strategies are evaluated each time we check if the feature toggle is enabled.
+The result of the Feature Toggle Strategy evaluation overrides the `enabled` status of the Feature Toggle.
+
+Simple Feature toggle check Example:
+
+```
+    class ClassWithQueryCancel
+    {
+        private final Supplier<Boolean> isQueryCancelEnabled;
+        private final Function<Object, Boolean> isQueryCancelEnabledForQueryId;
+
+        @Inject
+        public SupplierInjectionRunner(
+            @FeatureToggle("query-cancel") Supplier<Boolean> isQueryCancelEnabled,
+            @FeatureToggle("FunctionInjectionFeature") Function<Object, Boolean> isQueryCancelEnabledForQueryId)
+        {
+            this.isQueryCancelEnabledForQueryId = isQueryCancelEnabledForQueryId;
+            this.isQueryCancelEnabled = isQueryCancelEnabled;
+        }
+
+        /**
+        * simple check: checks enabled param of the configuration
+        */
+        public boolean testSimpleFeatureEnabled()
+        {
+            return isQueryCancelEnabled.get();
+        }
+
+        /**
+        * feature toggle check using a strategy that accepts input param
+        */
+         public boolean testFunctionInjectionFeatureEnabled(String queryId)
+        {
+            return isQueryCancelEnabledForQueryId.apply(queryId);
+        }
+    }
+```
+
+## Feature Toggle config file example
+
+Any change of the configuration in the configuration file (source) overrides the feature toggle configuration.
+Allowed configuration params are:
+
+- enable
+- strategy.<param>
```

Comment:
This line doesn't appear to display the way you seem to intend it to, see screenshot. 
<img width="977" alt="Screenshot 2024-07-08 at 11 40 28 AM" src="https://github.com/prestodb/presto/assets/7013443/e541a04b-8b92-4268-89bb-639389509799">